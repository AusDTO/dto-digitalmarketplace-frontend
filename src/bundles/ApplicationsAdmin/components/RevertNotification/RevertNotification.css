form textarea {
    height: 300px !important;
}

.form-button-wrapper {
    text-align: center
}