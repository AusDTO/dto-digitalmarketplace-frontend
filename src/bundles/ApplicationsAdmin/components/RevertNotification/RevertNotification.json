{
  "form_options": {},
  "assessments": {},
  "applications": {},
  "meta": {},
  "options": {}
}