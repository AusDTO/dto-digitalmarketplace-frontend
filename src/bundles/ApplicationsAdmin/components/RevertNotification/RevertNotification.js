import React, {Component} from 'react';
import {connect} from 'react-redux';
import {revertApplication} from '../../redux/modules/applications';
import './RevertNotification.css';
import {templateString} from '../../revertEmailTemplate';


export default class RevertedNotificationForm extends Component {

  static propTypes = {
    appID: React.PropTypes.number.isRequired,
    onClose: React.PropTypes.func.isRequired,
    defaultMessage: React.PropTypes.string.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      message: templateString
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({
      message: e.target.value
    })
  }

  resetMessageText(propsMessage) {
    this.setState({
      message: propsMessage
    });
  }

  render() {
    const {
      defaultMessage,
      appID,
      onClose,
      onRevertClick
    } = this.props;

    return (
      <form>
        <label htmlFor="formHeading">
          <h3>
            Revert application with optional email to seller
          </h3>
          <b>
            {`Provide a short clear reason, starting with a hyphen. Eg "${"- ABN missing a digit"}"`}
          </b>
          <textarea value={this.state.message} onChange={this.handleChange}/>
        </label>
        <div styleName="form-button-group">
          <div styleName="form-button-wrapper">
            <button
              type="button"
              style={{width: '200px'}}
              onClick={() => {
                onRevertClick(appID, this.state.message);
                this.resetMessageText(defaultMessage);
              }}>Revert & Send Email
            </button>
          </div>
          <div styleName="form-button-wrapper">
            <button
              type="button"
              style={{width: '200px'}}
              onClick={() => {
                onRevertClick(appID, '');
                this.resetMessageText(defaultMessage);
              }}>Revert Without Email
            </button>
          </div>
        </div>
        <a
          styleName="close-modal-prompt"
          tabIndex="0"
          onClick={() => {
              onClose(appID, this.state.message);
              this.resetMessageText(defaultMessage);
            }}
          onKeyUp={() => {
              onClose(appID, this.state.message);
              this.resetMessageText(defaultMessage);
            }}
          role="button">close
        </a>
      </form>
    )
  }
}


export const mapStateToProps = (state, ownProps) => {
  return {
    ...state,
    ...ownProps,
  }
};

const mapDispatchToProps = (dispatch, {onClose}) => {
  return {
    onRevertClick: (id, msg = '') => {
      onClose(id, msg);
      dispatch(revertApplication(id, msg));
    }
  }
};

export const ConnectedRevertedForm = connect(mapStateToProps, mapDispatchToProps)(RevertedNotificationForm);


