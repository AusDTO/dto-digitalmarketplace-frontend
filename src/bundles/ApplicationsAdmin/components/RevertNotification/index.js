import RevertNotification from './RevertNotification';

export default RevertNotification;