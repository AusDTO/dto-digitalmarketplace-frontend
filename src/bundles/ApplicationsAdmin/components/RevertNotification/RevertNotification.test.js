// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import {ConnectedRevertedForm} from './RevertNotification';
import renderer from 'react-test-renderer';

import createStore from '../../redux/create';
const sampleState = './RevertNotification.json'

test('RevertedNotificationForm with required attributes', () => {
  const store = createStore(Object.assign({},  sampleState));
  const component = renderer.create(
    <ConnectedRevertedForm
      store={store}
      message="email template default text"
      onClose={() => {return null}}
      appID={17}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

