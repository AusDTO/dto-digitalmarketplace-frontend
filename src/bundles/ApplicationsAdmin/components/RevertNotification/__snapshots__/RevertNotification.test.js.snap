// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevertedNotificationForm with required attributes 1`] = `
<form>
  <label
    htmlFor="formHeading"
  >
    <h3
      className="au-display-md"
    >
      Revert application with optional email to seller
    </h3>
    <b>
      Provide a short clear reason, starting with a hyphen. Eg "- ABN missing a digit"
    </b>
    <textarea
      onChange={[Function]}
      value="Dear Marketplace seller

Weâ€™ve reviewed your submission to the Digital Marketplace and require more detail or changes to your profile.

YOUR NOTES GO HERE

To update your profile:

  1. Log in to your [Digital Marketplace dashboard]({{ frontend_url }}/2/seller-dashboard).
  2. Update your seller profile.
  3. Resubmit your application for review.

We'll advise you once we've reviewed the changes.

Regards,  
The Digital Marketplace team  
Digital Transformation Agency  
[{{ frontend_url }}]({{ frontend_url }}) | [dta.gov.au](https://dta.gov.au)"
    />
  </label>
  <div
    styleName="form-button-group"
  >
    <div
      styleName="form-button-wrapper"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "width": "200px",
          }
        }
        type="button"
      >
        Revert & Send Email
      </button>
    </div>
    <div
      styleName="form-button-wrapper"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "width": "200px",
          }
        }
        type="button"
      >
        Revert Without Email
      </button>
    </div>
  </div>
  <a
    className="button"
    onClick={[Function]}
    onKeyUp={[Function]}
    styleName="close-modal-prompt"
    tabIndex="0"
  >
    close
  </a>
</form>
`;
